convert from CJS to ESM

dynst opened the pull request:
[Electron 28](https://www.electronjs.org/blog/electron-28-0) and newer has ESM support now.

dynst committed:
update to tailwindcss 3.4

ESM config files support was added in 3.3.0.

dynst committed:
convert asgardex package to ESM

dynst committed:
convert tailwind/postcss config files to ESM

Naq302 commented on package.json:
```diff
@@ -3,7 +3,8 @@
   "description": "WALLET AND EXCHANGE CLIENT FOR THORCHAIN, MAYACHAIN, CHAINFLIP",
-  "main": "build/main/electron.js",
+  "type": "module",
+  "exports": "build/main/electron.js",
```
When "type": "module", is added exports": "build/main/electron.js" shows a `String does not match the pattern of "^\./". warning. 

yarn dev fails
error during start dev server and electron app:
Error: No entry point found for electron app, please add a "main" field to package.json

Could be fixed via

```json
"main": "./build/main/electron.js",
  "exports": {
    ".": "./build/main/electron.js"
  },
  ```

dynst committed:
add missing vite config field renderer.resolve.extensions

It matters now that it's true ESM.

dynst committed:
fix electron-vite error

https://github.com/alex8088/electron-vite/blob/9ec164d33ed940b8fb979a8f3aedc12ffe51c51a/src/electron.ts#L14

Naq302 commented:
@dynst getting an error when doing a rebuild: 
```
start electron app...

App threw an error during load
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/asgardex-desktop/node_modules/fp-ts/lib/Either' imported from /asgardex-desktop/build/main/electron.js
Did you mean to import "fp-ts/lib/Either.js"?
A JavaScript error occurred in the main process
Uncaught Exception:
```
Not sure why as Either is imported and present. I assume they need to be imported differently in `electron.ts` but not sure.

dynst commented:
ESM `import` distinguishes between importing `/Either` and `/Either.js`, the filename extension is the difference there. But there shouldn't be a difference, if Vite is specifically configured to search for .ts and .js files if an extension is omitted:

https://github.com/asgardex/asgardex-desktop/blob/aae5c1bf1652c42bb7d68ac77fa8144ec0e8fa9c/electron.vite.config.mjs#L23

https://vite.dev/config/shared-options#resolve-extensions

But that field has a decent default, anyway, even if it wasn't set. And it's only applicable to internal files, not for external dependencies.

I guess we're not supposed to reach into the `/lib/` internal directory structure of a dependency. Instead we should:

`import { either as E } from 'fp-ts'`

There's a lot of import statements just like this one that bypass the module's official top-level `export` API, and they'll all need rewriting.
